core/ai.go

```diff
@@ -82,9 +83,51 @@ type AIModelConfig struct {
 	Currency      string  `json:"currency,omitempty"`
 }
 
+// UnmarshalJSON allows `PricePerUnit` to be specified as a string.
+func (s *AIModelConfig) UnmarshalJSON(data []byte) error {
```

Comment:
I see you cover different scenarios here, like `price_per_unit` can be a string or an int (or float?). Could you add some units for that to make sure it works for all these cases?